net/http2_conn_pool.go

```diff
@@ -0,0 +1,293 @@
+package net
+
+import (
+	"github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
+	"sync"
+	"sync/atomic"
+)
+
+// connPool is designed to be a speedy connection pool
+// Extensive testing for race conditions is a priority.
+// DO NOT edit without updating or checking against existing tests
+type connPool struct {
```

Comment:
nitpick: has this conn pool anything specific to do with http2, now? if not, the file name might need to change.

Reply:
Ah. No, nothing specific to http2. Will change the name

Reply:
Fixed here e16fd74141f5200dc28cb2ee3ebda6d41429f130